README.md

```diff
@@ -231,172 +257,271 @@ curl -s -i https://api.travis-ci.org/auth/github \
 
-## Writing build scripts
+## Creating a package
 
-Each version of each package has its own directory in `scripts/${package}/${version}`, e.g. [`scripts/libpng/1.6.28`](https://github.com/mapbox/mason/tree/master/scripts/libpng/1.6.28).
+### Prerequisites
 
-Below are details on ways to create packages for:
+Before getting started you should be able to answer the following questions.
 
- - new packages (not previously unpackaged software)
- - new versions of existing packages
- - header-only libraries and pre-compiled libraries
+**What are you packaging?**
 
-### Creating new packages
+There are different steps that you will need to follow depending on the type of library you are packaging. For example, there are fewer steps you need to take if you are creating a package of header-only code. Packaging compiled code has more steps because you'll need to tell Mason how to build your binaries. Another type of package is a [System package](#system-package) which has a unique process as well.
 
-If you are creating a package for previously unpackaged software start by creating a new directory for your `${package}/${version}` from within your mason checkout.
+**Are there previous versions already published?**
 
-For example if you want to name your package `yourlib` and it is version `1.0.0` you would do:
+Check the [list of packages](https://github.com/mapbox/mason/tree/master/scripts) to see if a previous version of your package exists. It's helpful to copy scripts from a previous version rather than creating code from scratch.
 
-```bash
-mkdir -p scripts/yourlib/1.0.0
-```
+**Where can Mason download your code?**
 
-The directory must contain two files:
+Every package needs to tell Mason where to download the code that it will build and publish, for example:
 
-    - `script.sh`
-    - `.travis.yml`
+ - `https://github.com/mapbox/geometry.hpp/archive/v0.9.2.tar.gz` for a Github release: [geometry 0.9.2](https://github.com/mapbox/geometry.hpp/releases/tag/v0.9.2)
+ - `https://github.com/mapbox/geometry.hpp/archive/b0e41cc5635ff8d50e7e1edb73cadf1d2a7ddc83.zip` for pre-release code hosted on Github: [geometry b0e41cc](https://github.com/mapbox/geometry.hpp/tree/b0e41cc5635ff8d50e7e1edb73cadf1d2a7ddc83)
 
-What you put in those files depend on what type of package you are creating. See:
+Note: Your code doesn't need to be hosted on Github in order for Mason to work. Your code can be hosted anywhere. Another common location is [SourceForge](#https://sourceforge.net/).
 
- - [Header-only package](#header-only-package)
- - [Pre-compiled library package](#pre-compiled-library-package)
+### Getting started
 
-### Creating new versions of packages
+These are just basic steps to help get you started. Depending on the complexity of building your code, you might have to review the [Script structure](#script-structure) section to get a better idea of how to further configure Mason to be able to create your package.
```

Comment:
@allieoop this `script structure` link doesn't seem to be working for me - perhaps the section title changed?

Reply:
good find. will update.